portfolio/src/main/webapp/calendar.js

```diff
@@ -151,15 +152,176 @@ function collectAllEvents() {
     const startTime = new Date(eventCardBody.childNodes[1].innerText);
     const endTime = new Date(eventCardBody.childNodes[2].innerText);
     const event = new CalendarEvent(eventName, startTime, endTime);
-    return event; 
+    return event;
   });
-
+ 
   return allEventsOnPage;
 }
-
+ 
 /** Checks if two calendar events are identical. */
 function eventsEqual(eventA, eventB) {
   return (eventA.name === eventB.name) &&
       (eventA.startTime.getTime() === eventB.startTime.getTime()) &&
       (eventA.endTime.getTime() === eventB.endTime.getTime());
 }
+
+/** Retrives the date that the user has picked for the scheduling. */
+function getUserPickedDate() {
+  const userPickedDate = $('#date-picker').val().split('-');
+  const year = userPickedDate[0];
+  const month = userPickedDate[1];
+  const date = userPickedDate[2];
+ 
+  const pickedDate = new Date();
+  pickedDate.setFullYear(year);
+  pickedDate.setMonth(month - 1);  // month is zero-indexed.
+  pickedDate.setDate(date);
+ 
+  return pickedDate;
+}
+ 
+/** Sets the date picker to the current date as the default. */
+function useTodayAsDefault() {
+  const today = new Date();
+  const year = today.getFullYear();
+  const month = today.getMonth() + 1;  // month is zero-indexed
+  const date = today.getDate();
+  
+  const todayString = parseTodayString(year, month, date);
+  $('#date-picker').val(todayString);
+}
+
+/** 
+ * Parses the current day's date into a dash separated string. 
+ * The parameters are of type String. 
+ */
+function parseTodayString(year, month, date) {
+  // Adds leading 0 as padding for month and date strings.
+  if (parseInt(month) < 10) {
+    month = '0' + month;
+  }
+  
+  if (parseInt(date) < 10) {
+    date = '0' + date;
+  }
+  const todayString = year + '-' + month + '-' + date;
+  return todayString; 
+}
+ 
+/**
+ * Sets the default calendar event start and times.
+ * The default start time is set to the closet hour.
+ * The default end time is set to the end of working hour.
+ */
+function setClosestEventTime() {
```

Comment:
Consider renaming to setNewEventStartAndEndTimes or something to clarify the side effect of this method.

Reply:
Updated.